readthedocs/projects/views/public.py

```diff
@@ -421,6 +427,7 @@ def project_versions(request, project_slug):
             'inactive_versions': inactive_versions,
             'active_versions': active_versions,
             'project': project,
+            'is_project_admin': AdminPermission.is_admin(request.user, project),
```

Comment:
I wonder if its better to add some kind of caching to `is_admin` so we don't hit this over and over? We could even just cache it for like 10 seconds or something, but that might make the code a bit simpler than having an explicit cache in the template?

Reply:
Hmm, it's a class method, the cache key would be something like the class, the object id and the user. I'll try and see how that would work.

Reply:
Ok, I did a test. Caching reduces the number of queries to almost the same number, but it adds more overhead in the caching side (more CPU and time), making the requests slow.

![Screenshot_2020-07-08 Versions - Read the Docs for Business(1)](https://user-images.githubusercontent.com/4975310/86983376-53c2ab80-c151-11ea-9879-bf89e2d92e9f.png)


![Screenshot_2020-07-08 Versions - Read the Docs for Business](https://user-images.githubusercontent.com/4975310/86983336-42799f00-c151-11ea-9721-8d09a574608c.png)

I can submit a PR with the cache, so cache + this PR we should reduce more the slowness

Reply:
Ah yea, good point on all the cache calls -- I think this approach is probably best 👍